assets/RNAseq_report.Rmd

```diff
@@ -1711,36 +1711,41 @@ The table below provides more detail on all enriched pathways."))
         #Reading DE genes list
         fname <- tools::file_path_sans_ext(basename(file))
         DE_genes <- read.csv(file = paste0("differential_gene_expression/DE_genes_tables/", file), sep="\t", header = T)
+
         if(nrow(DE_genes) <= 1){
             next
         }
         q = as.character(DE_genes$Ensembl_ID)
         q_list <- append(q_list, list(q))
         q_names <- append(q_names, fname)
+
     }
     names(q_list) <- q_names
+    if (length(q_list) > 0) {
 
-    #gost query
-    gostres <- gost(query=q_list,
+        #gost query
+        gostres <- gost(query=q_list,
                     organism=tolower(params$organism),
                     significant=T,
                     correction_method="fdr",
                     sources=datasources,
                     user_threshold=params$pval_threshold)
-    path_enrich <- as.data.frame(gostres$result)
-    if (nrow(path_enrich) > 0){
-        pg2 <- gostplot(gostres, capped=T, interactive=T)
-        pg2[['x']][['layout']][['annotations']][[1]][['x']] <- -params$pval_threshold
-
-        #limit gostplot y maximum
-        for (counter in c(1:length(contrast_files))) {
-            pg2[['x']][['layout']][[paste0('yaxis', ifelse(counter>1, counter, ""))]][['range']] <- c(0,ceiling(2*-log10(max(gostres$p_value))))
-        }
+        path_enrich <- as.data.frame(gostres$result)
+        if (nrow(path_enrich) > 0){
+            pg2 <- gostplot(gostres, capped=T, interactive=T)
+            pg2[['x']][['layout']][['annotations']][[1]][['x']] <- -params$pval_threshold
+
+            #limit gostplot y maximum
+            for (counter in c(1:length(contrast_files))) {
+                pg2[['x']][['layout']][[paste0('yaxis', ifelse(counter>1, counter, ""))]][['range']] <- c(0,ceiling(2*-log10(max(gostres$p_value))))
+            }
 
-        #add export button to plot
-        config(pg2, modeBarButtonsToAdd = list(svg_exp)) %>% layout(width=500, height = 400*length(contrast_files))
-    } else {
-        cat("No enriched pathways were found.")
+            #add export button to plot
+            config(pg2, modeBarButtonsToAdd = list(svg_exp)) %>% layout(width=500, height = 400*length(contrast_files))
+        } else {
+                cat("No enriched pathways were found.")
+                path_enrich=data.frame(list())
+        }
     }
```

Comment:
It would be nice to generate a message here as well in case there are no enriched pathways, in the same way as in the previous clause